services/web/client/source/resource/form/service.json

```diff
@@ -0,0 +1,186 @@
+                },
+                "lastName": {
+                    "type": "string",
+                    "title": "Last name"
+                },
+                "email": {
+                    "type": "string",
+                    "title": "Email"
+                },
+                "affiliation": {
+                    "type": "string",
+                    "title": "Affiliation"
+                }
+            }
+        },
+        "serviceInfo": {
+            "type": "object",
+            "title": "Service information",
+            "required": [
+                "description", 
+                "commandLine"
+            ],
+            "properties": {
+                "description": {
+                    "type": "string",
+                    "title": "Description of the service",
+                    "description": "Please provide with detailed information about how to run the service and any dependencies required."
+                },
+                "commandLine": {
+                    "type": "string",
+                    "title": "Command line interface to run service",
+                    "description": "In the command line, what command is used to run the code? E.g. In Matlab Command Window: 'myproject(input1, input2, input3)'"
+                },
+                "url": {
+                    "type": "string",
+                    "title": "URL of the repository"
+                },
+                "codeVariant": {
+                    "type": "string",
+                    "title": "Language/Code type",
+                    "description": "Choose one of the accepted options.",
+                    "enum": ["matlab", "opencor", "python"]
+                },
+                "files": {
+                    "type": "string",
+                    "title": "File upload",
+                    "description": "If your code doesn't live inside a repository, you can upload a single .zip file containing it. Maximum size is 10MB",
+                    "contentMediaType": "application/zip"
+                }
+            }
+        },
+        "serviceInterface": {
+            "type": "object",
+            "title": "Service inputs and outputs",
+            "properties": {
+                "inputs": {
+                    "title": "Inputs",
+                    "type": "array",
+                    "items": {
+                        "type": "object",
+                        "title": "input",
+                        "required": [
+                            "name",
+                            "type",
+                            "label",
+                            "defaultValue",
+                            "description"
+                        ],
+                        "properties": {
+                            "name": {
+                                "type": "string",
+                                "title": "Name",
+                                "description": "Input's ID (variable name)."
```

Comment:
and also maybe state that this one must be unique.

Reply:
ok, I will add this in the text

Reply:
could you also add "with units" in the description - since you are changing anyway =)
"description": "Input's ID (variable name), its meaning and its units, if applicable"

Reply:
that is just one field for giving a name for that input. the unit has it's own field... and the meaning I guess it goes into the description field
![Screenshot from 2020-05-05 13-39-20](https://user-images.githubusercontent.com/4764217/81062206-dc189c00-8ed5-11ea-976c-625249de6481.png)

Reply:
there is no unit field. do you want me to add it?

Reply:
in fact that would be very nice (allowing "not applicable" to be an option  if you are doing verification). Thank you!

Reply:
If you want a list of options and then a combo box, then I would need the list of possible units and I don't thinks that is feasible or convenient... I will leave it as it is from now. You can modify the schema anytime!